src/geff/metadata_schema.py

```diff
@@ -208,6 +255,37 @@ class GeffMetadata(BaseModel):
         "Each axis can additionally optionally define a `unit` key, which should match the valid"
         "OME-Zarr units, and `min` and `max` keys to define the range of the axis.",
     )
+
+    # PROPOSITION 1: separate metadata by component type
+    node_props_metadata: dict[str, PropMetadata] | None = Field(
+        None,
+        description=(
+            "Metadata for node properties. The keys are the property identifiers, "
+            "and the values are PropMetadata objects describing the properties."
+        ),
+    )
+    edge_props_metadata: dict[str, PropMetadata] | None = Field(
+        None,
+        description=(
+            "Metadata for edge properties. The keys are the property identifiers, "
+            "and the values are PropMetadata objects describing the properties."
+        ),
+    )
+    tracklet_props_metadata: dict[str, PropMetadata] | None = Field(
+        None,
+        description=(
+            "Metadata for tracklet properties. The keys are the property identifiers, "
+            "and the values are PropMetadata objects describing the properties."
+        ),
+    )
+    lineage_props_metadata: dict[str, PropMetadata] | None = Field(
+        None,
+        description=(
+            "Metadata for lineage properties. The keys are the property identifiers, "
+            "and the values are PropMetadata objects describing the properties."
+        ),
+    )
```

Comment:
Let's leave these out for now since we haven't formalized the tracklet and lineage property data structure yet. We can follow this pattern when that's ready to go in. 
```suggestion
    tracklet_props_metadata: dict[str, PropMetadata] | None = Field(
        None,
        description=(
            "Metadata for tracklet properties. The keys are the property identifiers, "
            "and the values are PropMetadata objects describing the properties."
        ),
    )
    lineage_props_metadata: dict[str, PropMetadata] | None = Field(
        None,
        description=(
            "Metadata for lineage properties. The keys are the property identifiers, "
            "and the values are PropMetadata objects describing the properties."
        ),
    )
```